kate_brownbill_dungon_fighters_v1.0.py

```diff
@@ -0,0 +1,838 @@
+    return (non_acv_team, non_acv_team_names)
+    
+def ai_swap(team, acv_fighter):
+    non_acv_team = []
+    for i in team:
+        if i != acv_fighter and i.state != 0:
+            non_acv_team.append(i)
+    fighter = non_acv_team[0]
+    return fighter
+
+def get_info(): #get's input from the player on what info they want
+    menu = get_menu(["back" ,"my team", "my fighter", "enemy fighter", "enemy player", "moves"], "What do you want to learn about?\n[BACK] [MY TEAM] [MY FIGHTER] [ENEMY FIGHTER] [ENEMY PLAYER] [MOVES]\n", "You can't get info for that.")
+    return menu
+    
+def create_prompt(lst): #creates a menu prompt string like so: [ITEM 1] [ITEM 2] [ITEM 3]...\
+    prompt = ""
+    for i in lst:
+        prompt = prompt + " [" + str(i).upper() + "]"
+    return prompt
+
+def create_moves(number):
+    moves = [FIREBOLT, FRWD_SWNG, MAGIC_RFCT, MAGIC_SWNG, SHIELD_BASH]
+    random.shuffle(moves)
+    set = []
+    for i in range(number):
+        set.append(moves.pop())
+    return set
+
+#runs check functions depending on menu's value
+def do_info(menu, your_team, your_fighter, enemy_fighter, enemy_player):
+    if menu == 1:
+        print("Your team:")
+        check_team(your_team)
+    elif menu == 2:
+        print("Your fighter:")
+        check_fighter(your_fighter)
+    elif menu == 3:
+        print("Enemy fighter:")
+        check_fighter(enemy_fighter)
+    elif menu == 4:
+        print("Enemy player:")
+        check_player(enemy_player)
+    elif menu == 5:
+        print("Moves:")
+        check_moves()
+
+#prints importent info about a fighter
+def check_fighter(fighter):
+        print("Name: " + fighter.name)
+        print(fighter.ft)
+        print("HP: " + str(fighter.hp))
+        print("Attack: " + str(fighter.attack))
+        print("Defence: " + str(fighter.defence))
+        m = ""
+        for i in fighter.moves:
+            m = m + " " + move_to_txt[i] + ","
+        m = m[:-1]
+        print("Moves:" + m)
+
+#prints importent info about a team
+def check_team(team):
+    for i in team:
+        print("===============")
+        check_fighter(i)
+        if i.state == 0:
+            print("Unconscious")
+        if i.state == 1:
+            print("Conscious")
+        
+
+#prints importent info about a player
+def check_player(player):
+    print("Name: " + player.name)
+    print(player.ft)
+    print("Player type: " + user_to_txt[player.user])
+
+def check_moves():
+    print("Firebolt is good agesnt foraward swing and magic swing.")
+    print("Forward swing is good agenst shield bash and magic reflect.")
+    print("Magic reflect is good agenst firebolt and magic swing.")
+    print("Magic swing is good agenst forward swing and shield bash.")
+    print("Shield bash is good agenst firebolt and magic reflect.")
+
+#checks to see if any members on the team are alive
+def test_alive(team):
+    alive = False
+    for i in team:
+        #print i.state
+        if i.state != 0:
```

Comment:
This is a place where the code is easier to reason about if you rename state to __state to tell people that outside functions don't have direct access and you add, to play,

```
def get_state(self)
   return self.__state
```

The problem that I have, as a reader or future maintainer, is that I now have to worry about ~1000 lines of code that can access or modify state anywhere.